doc/rm/c_cpp_coding_style.md

```diff
@@ -173,6 +186,49 @@ This may change if we find cases where this initialization improves readability.
 When initializing an array, initializers *may* be designated when that makes the array more readable (e.g., lookup tables that are mostly zeroed). Mixing designated and undesignated initializers, or using nested initializers, is still
 forbidden.
 
+### Nonstandard Attributes
+
+The following nonstandard attributes may be used:
+*   `section(<name>)` to put a definition into a given object section.
+*   `weak` to make a symbol definition have weak linkage.
+*   `naked` to ensure a function has no prolog or epilog.
+*   `packed` to ensure a struct contains no padding.
+
+It is recommended that other nonstandard attributes are not used.
+
+All nonstandard attributes must be supported by both GCC and Clang.
+
+***The use of nonstandard attributes must be hidden by a macro definition starting `OT_`.***
+
+For instance, `__attribute__((section("foo")))` should not be used on a declaration.
+Instead, the attribute should be hidden behind a macro, like below.
+```c
+#define OT_SECTION(name) __attribute__((section(name)))
+//...
+void _crt(void) OT_SECTION(".crt");
```

Comment:
Should we specify where the attributes go? I would actually suggest
```c
OT_SECTION(".crt")
void _crt(void);
```
This has the advantages of:
- Working regardless of being applied to a function prototype or a function definition.
- It helps stop the line width from getting totally unreasonable.

Mind, we don't really need to put it on its own line, but at least putting it at the start is a good idea.